### PR TITLE
codegen: Add an option to skip comment generation.

### PR DESCRIPTION
**Pull request by emilio**
This is mostly a work around https://github.com/servo/rust-bindgen/issues/426,
until we implement the proper fix.

r? @fitzgen

### CODE REVIEW

File: src/codegen/mod.rs [hunk 1]
```diff
@@ -1007,8 +1011,10 @@ impl CodeGenerator for CompInfo {
             };
 
             let mut attrs = vec![];
-            if let Some(comment) = field.comment() {
-                attrs.push(attributes::doc(comment));
+            if ctx.options().generate_comments {
```
Comment (fitzgen) [inline src/codegen/mod.rs]: This is a whole lot of copy paste... What if we made `Item::comment` and `Field::comment` take the context and check themselves? The earlier we push this check, we could even potentially avoid keeping the comments around in memory.

I would be fine with landing this code now and investigating this later, but if you do want to land the code now, please file an issue and maybe mark it as a good first bug to attract new contributors.

File: Cargo.lock [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [root]
 name = "bindgen"
-version = "0.20.2"
+version = "0.20.3"
```
Comment (fitzgen) [inline Cargo.lock]: This should ideally be part of the previous commit.

File: src/lib.rs [hunk 3]
```diff
@@ -188,6 +188,15 @@ impl Builder {
         self
     }
 
+    /// Whether to whitelist types recursively or not. Defaults to true.
+    ///
+    /// See https://github.com/servo/rust-bindgen/issues/429 for different use
+    /// cases for this.
```
Comment (fitzgen) [inline src/lib.rs]: I think we should elaborate the use cases here: one shouldn't have to jump from the rustdocs to a random github issue to understand why a method exists and to figure out if they should use it or not.

File: Cargo.lock (hunk 2 above)
Comment (emilio) [inline Cargo.lock]: Right, moved.

File: src/lib.rs (hunk 3 above)
Comment (emilio) [inline src/lib.rs]: Fair enough :)

File: src/codegen/mod.rs (hunk 1 above)
Comment (emilio) [inline src/codegen/mod.rs]: That's right, but I thought that the alternative would be less readable? Anyway, will file.

**Comment by cjhowe** [comment]
This is much appreciated. Thanks!

**Pull request merged**